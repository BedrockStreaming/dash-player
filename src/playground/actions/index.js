export const HANDLE_PLAYER_EVENT = 'HANDLE_PLAYER_EVENT';
export const HANDLE_SOURCE_CHANGE = 'HANDLE_SOURCE_CHANGE';

export const handlePlayerEvent = event => ({ type: HANDLE_PLAYER_EVENT, event });
export const handleSourceChange = source => ({ type: HANDLE_SOURCE_CHANGE, source });
