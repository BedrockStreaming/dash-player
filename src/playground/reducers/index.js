import { HANDLE_PLAYER_EVENT, HANDLE_SOURCE_CHANGE } from '../actions';

const DEFAULT_SOURCE = 'http://www.bok.net/dash/tears_of_steel/cleartext/stream.mpd';

export const initialState = {
  events: [],
  source: DEFAULT_SOURCE,
};

export const playgroundReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case HANDLE_PLAYER_EVENT:
      return { ...state, events: [...state.events, action.event] };

    case HANDLE_SOURCE_CHANGE:
      return { ...state, source: action.source, events: [] };

    default:
      return state;
  }
};
