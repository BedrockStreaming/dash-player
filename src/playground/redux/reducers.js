import { combineReducers } from 'redux';
import { playgroundReducer } from '../reducers';

export const combinedReducers = combineReducers({ playground: playgroundReducer });
