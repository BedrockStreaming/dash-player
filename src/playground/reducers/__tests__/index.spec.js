import deepFreeze from 'deep-freeze';
import { playgroundReducer } from '../../reducers';

import { HANDLE_PLAYER_EVENT, HANDLE_SOURCE_CHANGE } from '../../actions';

jest.unmock('../../reducers');

function freezedReducer(state, action) {
  return playgroundReducer(deepFreeze(state), action);
}

describe('playgroundReducer', () => {
  const initialState = {
    events: [],
    source: 'http://www.bok.net/dash/tears_of_steel/cleartext/stream.mpd',
  };

  it('should HANDLE_PLAYER_EVENT', () => {
    const action = { type: HANDLE_PLAYER_EVENT, event: {} };

    expect(freezedReducer(initialState, action)).toMatchSnapshot();
  });

  it('should SEARCH_VI', () => {
    const action = { type: HANDLE_SOURCE_CHANGE, source: 'source' };

    expect(freezedReducer(initialState, action)).toMatchSnapshot();
  });
});
