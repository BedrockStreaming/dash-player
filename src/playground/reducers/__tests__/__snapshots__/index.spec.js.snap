// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playgroundReducer should HANDLE_PLAYER_EVENT 1`] = `
Object {
  "events": Array [
    Object {},
  ],
  "source": "http://www.bok.net/dash/tears_of_steel/cleartext/stream.mpd",
}
`;

exports[`playgroundReducer should SEARCH_VI 1`] = `
Object {
  "events": Array [],
  "source": "source",
}
`;
